core/src/main/scala/com/databricks/labs/remorph/RootTableIdentifier.scala

```diff
@@ -0,0 +1,51 @@
+package com.databricks.labs.remorph
+
+import com.databricks.labs.remorph.parsers.{intermediate => ir}
+import com.databricks.labs.remorph.transpilers.TranspilerFactory
+
+class RootTableIdentifier(engine: String) {
+
+  private val transpiler = TranspilerFactory.getTranspiler(engine)
```

Comment:
```suggestion
class RootTableIdentifier(Transpiler transpiler) {
```
it doesn't seem that you fully understood my previous comments about dependency injection. let's prohibit any use of Factory pattern other than ContextFactory, because i don't see the pattern used in an effective way.

this code design is bad, as one is coupled to invariants defined by a factory class and nothing else than that. what if we want to test with transpiler that always fails only on second query?

Reply:
Made changes.

Reply:
Took care.